src/strings/ops.c

```diff
@@ -328,9 +328,15 @@ static void iterate_gi_into_string(MVMThreadContext *tc, MVMGraphemeIter *gi, MV
     }
     else {
         MVMStringIndex result_pos = 0;
-        result->body.storage_type            = MVM_STRING_GRAPHEME_32;
-        result32 = result->body.storage.blob_32 =
-            result_graphs ? MVM_malloc(result_graphs * sizeof(MVMGrapheme32)) : NULL;
+        if (result_graphs <= 2) {
+            result->body.storage_type            = MVM_STRING_IN_SITU_32;
+            result32 = result->body.storage.in_situ_32;
+        }
+        else {
+            result->body.storage_type            = MVM_STRING_GRAPHEME_32;
+            result32 = result->body.storage.blob_32 =
+                result_graphs ? MVM_malloc(result_graphs * sizeof(MVMGrapheme32)) : NULL;
```

Comment:
i think the ternary is not necessary since result_graphs is already checked to be > 2 in this branch, and [further up](https://github.com/MoarVM/MoarVM/pull/1825/files#diff-bc76be5614df2473912916db269fc246c79efbc83a0a065ee8b785aa3c68d040R258) we already bail out early if result_graphs is zero

Reply:
Yeah, good catch, I'll make the change.